server/roomlogs.js

```diff
@@ -90,10 +90,11 @@ class Roomlog {
 		log = [];
 		for (let i = 0; i < this.log.length; ++i) {
 			const line = this.log[i];
-			if (line === '|split') {
-				const ownLine = this.log[i + channel + 1];
+			const split = /\|split\|p(\d)/g.exec(line);
+			if (split) {
+				const ownLine = this.log[i + (Number(split[0]) === channel ? 1 : 2)];
```

Comment:
What does `channel` mean here? The previous mapping was `0:spectator, 1:p1, 2:p2, 3:omniscient`, I think, but this refactor should probably change that mapping.

Reply:
true, I'm neglecting `omniscient`. I'm assuming its now 0 - 5.